components/clarinet-deployments/src/onchain/mod.rs

```diff
@@ -666,9 +666,18 @@ pub fn apply_on_chain_deployment(
 
                     // Remapping principals - This is happening
                     let mut source = tx.source.clone();
-                    for (src_principal, dst_principal) in tx.remap_principals.iter() {
-                        let src = src_principal.to_address();
-                        let dst = dst_principal.to_address();
+                    for (src_principal, dst_principal) in tx
+                        .remap_principals
+                        .iter()
+                        .map(|(src, dst)| (src.to_address(), dst.to_address()))
+                        .chain(
+                            contracts_ids_to_remap
+                                .iter()
+                                .map(|(k, v)| (k.clone(), v.clone())),
+                        )
```

Comment:
Why do ContractIds need to be remapped here, but didn't need to before this PR?

Reply:
I think they should have been. This is needed to handle requirements in the requirements. In this case, the SIP-009 and 010 traits in the subnet contract. My guess is that this scenario just didn't come up before.